integration-tests/contract-security-zero-transfer.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { CONFIGS } from './config';
+import { RpcClient } from '@taquito/rpc';
+
+// TC-007 - A 0tez transaction to an implicit account should fail.
+
+const client = new RpcClient(' https://ithacanet.ecadinfra.com');
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+
+  describe(`Test contracts using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+
+      done();
+    });
+
+    it('Verify that Transactions of 0ꜩ towards a contract without code are forbidden', async () => {
+      try {
+        const op = await Tezos.contract.originate({
+          code: `{ parameter address ;
+                      storage unit ;
+                      code { UNPAIR ;
+                             CONTRACT unit ;
+                             IF_NONE { PUSH string "none" ; FAILWITH } {} ;
+                             PUSH mutez 0 ;
+                             UNIT ;
+                             TRANSFER_TOKENS ;
+                             SWAP ;
+                             NIL operation ;
+                             DIG 2 ;
+                             CONS ;
+                             PAIR } }
+                          `,
+          init: { prim: 'Unit' },
+        });
+
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const contract = await op.contract();
+        expect(await contract.storage()).toBeTruthy();
+
+         const publicKeyHash = await Tezos.signer.publicKeyHash();
+         const opTransfer = await Tezos.contract.transfer({ to: publicKeyHash, amount: 0 });
+         await opTransfer.confirmation();
+      } catch (error: any) {
```

Comment:
The origination and transfer are in the same try/catch. Which one is the failing operation? If both, should they be slipt in 2 distinct try/catch?

Reply:
fixed this thanks

Reply:
THis has been ocrrected. Thanks.